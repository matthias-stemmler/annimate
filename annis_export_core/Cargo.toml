[package]
name = "annis_export_core"
version = "0.1.0"
edition = "2021"

[dependencies]
csv = "1"
graphannis = "3"
graphannis-core = "3"
itertools = "0.12"
regex = "1"
thiserror = "1"

[dev-dependencies]
indoc = "2"
insta = { version = "1", features = ["serde"] }
serde = { version = "1", features = ["derive"] }
tempfile = "3"
