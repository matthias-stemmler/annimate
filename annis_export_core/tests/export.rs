use annis_export_core::{CorpusStorage, ExportFormat, QueryConfig, QueryLanguage};
use serde::Serialize;
use std::fs::File;

macro_rules! export_test {
    ($(
        $name:ident: {
            corpus_path: $corpus_path:expr,
            corpus_name: $corpus_name:expr,
            aql_query: $aql_query:expr,
            query_config: {
                left_context: $left_context:expr,
                right_context: $right_context:expr,
                query_language: $query_language:expr,
                segmentation: $segmentation:expr,
            },
        }
    )*) => {
        $(
            #[test]
            fn $name() {
                let test_data = TestData {
                    corpus_path: $corpus_path,
                    corpus_name: $corpus_name,
                    aql_query: $aql_query,
                    query_config: TestQueryConfig {
                        left_context: $left_context,
                        right_context: $right_context,
                        query_language: {
                            use QueryLanguage::*;
                            $query_language
                        },
                        segmentation: $segmentation,
                    }
                };

                let storage = CorpusStorage::from_db_dir(tempfile::tempdir().unwrap()).unwrap();

                storage
                    .import_corpora_from_zip(
                        File::open(concat!(env!("CARGO_MANIFEST_DIR"), "/tests/data/", $corpus_path)).unwrap(),
                        |_| (),
                    )
                    .unwrap();

                let mut export_bytes = Vec::new();

                storage
                    .export_matches(
                        $corpus_name,
                        $aql_query,
                        QueryConfig {
                            left_context: $left_context,
                            right_context: $right_context,
                            query_language: {
                                use QueryLanguage::*;
                                $query_language
                            },
                            segmentation: $segmentation.map(|s: &str| s.to_string()),
                        },
                        ExportFormat::Csv,
                        &mut export_bytes,
                        |_| (),
                    )
                    .unwrap();

                let export = String::from_utf8(export_bytes).unwrap();

                insta::with_settings!(
                    {
                         info => &test_data,
                         omit_expression => true,
                    },
                    { insta::assert_snapshot!(export) }
                );
            }
        )*
    };
}

export_test! {
    subtok_segmentation_tokens: {
        corpus_path: "subtok.demo_relANNIS.zip",
        corpus_name: "subtok.demo",
        aql_query: "pos=\"DT\"",
        query_config: {
            left_context: 4,
            right_context: 4,
            query_language: AQL,
            segmentation: None,
        },
    }
    subtok_segmentation_diplomatic: {
        corpus_path: "subtok.demo_relANNIS.zip",
        corpus_name: "subtok.demo",
        aql_query: "pos=\"DT\"",
        query_config: {
            left_context: 4,
            right_context: 4,
            query_language: AQL,
            segmentation: Some("diplomatic"),
        },
    }
    subtok_segmentation_norm: {
        corpus_path: "subtok.demo_relANNIS.zip",
        corpus_name: "subtok.demo",
        aql_query: "pos=\"DT\"",
        query_config: {
            left_context: 4,
            right_context: 4,
            query_language: AQL,
            segmentation: Some("norm"),
        },
    }
    subtok_gap: {
        corpus_path: "subtok.demo_relANNIS.zip",
        corpus_name: "subtok.demo",
        aql_query: "pos=\"DT\" .5,5 pos=\"DT\"",
        query_config: {
            left_context: 1,
            right_context: 1,
            query_language: AQL,
            segmentation: None,
        },
    }
    pcc2: {
        corpus_path: "pcc2_v7_relANNIS.zip",
        corpus_name: "pcc2",
        aql_query: "Sent _i_ NP",
        query_config: {
            left_context: 10,
            right_context: 10,
            query_language: AQL,
            segmentation: None,
        },
    }
}

#[derive(Serialize)]
struct TestData {
    corpus_path: &'static str,
    corpus_name: &'static str,
    aql_query: &'static str,
    query_config: TestQueryConfig,
}

#[derive(Serialize)]
struct TestQueryConfig {
    left_context: usize,
    right_context: usize,
    query_language: QueryLanguage,
    segmentation: Option<&'static str>,
}
