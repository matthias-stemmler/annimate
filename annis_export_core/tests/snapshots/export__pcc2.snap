---
source: annis_export_core/tests/export.rs
info:
  corpus_path: pcc2_v7_relANNIS.zip
  corpus_name: pcc2
  aql_query: Sent _o_ NP
  query_config:
    left_context: 10
    right_context: 10
    query_language: AQL
    segmentation: ~
---
Number,Document,Context 1,Match 1,Context 2,Match 2,Context 3,Match 3,Context 4
1,pcc2/11299,Feigenblatt,Die Jugendlichen,,in Zossen wollen ein Musikcafé .,Das forderten sie bei der ersten Zossener Runde am Dienstagabend,,
2,pcc2/11299,Feigenblatt,Die Jugendlichen in,,Zossen,,wollen ein Musikcafé .,Das forderten sie bei der ersten Zossener Runde am Dienstagabend
3,pcc2/11299,Feigenblatt,Die Jugendlichen in Zossen wollen,,ein Musikcafé,,.,Das forderten sie bei der ersten Zossener Runde am Dienstagabend
4,pcc2/11299,Feigenblatt Die Jugendlichen in Zossen wollen ein Musikcafé .,Das,,forderten sie bei der ersten Zossener Runde am Dienstagabend .,"Dass die Politiker der Stadt dafür Verständnis haben , ist",,
5,pcc2/11299,Feigenblatt Die Jugendlichen in Zossen wollen ein Musikcafé .,Das forderten,,sie,,bei der ersten Zossener Runde am Dienstagabend .,"Dass die Politiker der Stadt dafür Verständnis haben , ist"
6,pcc2/11299,Feigenblatt Die Jugendlichen in Zossen wollen ein Musikcafé .,Das forderten sie bei,,der ersten Zossener Runde,,am Dienstagabend .,"Dass die Politiker der Stadt dafür Verständnis haben , ist"
7,pcc2/11299,Feigenblatt Die Jugendlichen in Zossen wollen ein Musikcafé .,Das forderten sie bei der ersten Zossener Runde am,,Dienstagabend,,.,"Dass die Politiker der Stadt dafür Verständnis haben , ist"
8,pcc2/11299,forderten sie bei der ersten Zossener Runde am Dienstagabend .,Dass,,die Politiker,,"der Stadt dafür Verständnis haben , ist löblich .",Mit dem Treffen im Rathaus ist somit auch ein Dialog
9,pcc2/11299,forderten sie bei der ersten Zossener Runde am Dienstagabend .,Dass die Politiker,,der Stadt,,"dafür Verständnis haben , ist löblich .",Mit dem Treffen im Rathaus ist somit auch ein Dialog
10,pcc2/11299,forderten sie bei der ersten Zossener Runde am Dienstagabend .,Dass die Politiker der Stadt,,dafür,,"Verständnis haben , ist löblich .",Mit dem Treffen im Rathaus ist somit auch ein Dialog
11,pcc2/11299,forderten sie bei der ersten Zossener Runde am Dienstagabend .,Dass die Politiker der Stadt dafür,,Verständnis,,"haben , ist löblich .",Mit dem Treffen im Rathaus ist somit auch ein Dialog
12,pcc2/11299,"Politiker der Stadt dafür Verständnis haben , ist löblich .",Mit,,dem Treffen,,im Rathaus ist somit auch ein Dialog zwischen den Generationen angestoßen .,Dass die beiden geladenen Jugendlichen im Laufe des Abends immer
13,pcc2/11299,"Politiker der Stadt dafür Verständnis haben , ist löblich .",Mit dem Treffen im,,Rathaus,,ist somit auch ein Dialog zwischen den Generationen angestoßen .,Dass die beiden geladenen Jugendlichen im Laufe des Abends immer
14,pcc2/11299,"Politiker der Stadt dafür Verständnis haben , ist löblich .",Mit dem Treffen im Rathaus ist somit auch,,ein Dialog,,zwischen den Generationen angestoßen .,Dass die beiden geladenen Jugendlichen im Laufe des Abends immer
15,pcc2/11299,"Politiker der Stadt dafür Verständnis haben , ist löblich .",Mit dem Treffen im Rathaus ist somit auch ein Dialog zwischen,,den Generationen,,angestoßen .,Dass die beiden geladenen Jugendlichen im Laufe des Abends immer
16,pcc2/11299,ist somit auch ein Dialog zwischen den Generationen angestoßen .,Dass,,die beiden geladenen Jugendlichen,,"im Laufe des Abends immer weniger zu Wort kamen , war sicher keine böse Absicht , ärgerlich ist es trotzdem .","Und aberwitzig dazu . Es entbehrt nicht der Komik ,"
17,pcc2/11299,ist somit auch ein Dialog zwischen den Generationen angestoßen .,Dass die beiden geladenen Jugendlichen im,,Laufe,,"des Abends immer weniger zu Wort kamen , war sicher keine böse Absicht , ärgerlich ist es trotzdem .","Und aberwitzig dazu . Es entbehrt nicht der Komik ,"
18,pcc2/11299,ist somit auch ein Dialog zwischen den Generationen angestoßen .,Dass die beiden geladenen Jugendlichen im Laufe,,des Abends,,"immer weniger zu Wort kamen , war sicher keine böse Absicht , ärgerlich ist es trotzdem .","Und aberwitzig dazu . Es entbehrt nicht der Komik ,"
19,pcc2/11299,ist somit auch ein Dialog zwischen den Generationen angestoßen .,Dass die beiden geladenen Jugendlichen im Laufe des Abends immer weniger zu,,Wort,,"kamen , war sicher keine böse Absicht , ärgerlich ist es trotzdem .","Und aberwitzig dazu . Es entbehrt nicht der Komik ,"
20,pcc2/11299,ist somit auch ein Dialog zwischen den Generationen angestoßen .,"Dass die beiden geladenen Jugendlichen im Laufe des Abends immer weniger zu Wort kamen , war sicher",,keine böse Absicht,,", ärgerlich ist es trotzdem .","Und aberwitzig dazu . Es entbehrt nicht der Komik ,"
21,pcc2/11299,ist somit auch ein Dialog zwischen den Generationen angestoßen .,"Dass die beiden geladenen Jugendlichen im Laufe des Abends immer weniger zu Wort kamen , war sicher keine böse Absicht , ärgerlich ist",,es,,trotzdem .,"Und aberwitzig dazu . Es entbehrt nicht der Komik ,"
22,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .",Es,,"entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die",,
23,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .",Es entbehrt nicht,,der Komik,,", wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
24,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich",,drei Erwachsene,,"- Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
25,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene -",,Karola Andrae,,"( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
26,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae",,( Bürgerbündnis/FDP ),,", Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
27,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) ,",,Susanne Michler,,"( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
28,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler",,( CDU ),,"und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
29,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und",,Joachim Zanow,,"( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
30,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow",,( SPD ),,"- darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
31,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was",,Jugendliche,,"wollen und brauchen , ohne auf die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
32,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf",,die Idee,,"zu kommen , sie selbst zu fragen .","Und das , obwohl sie ihnen gegenüber sitzen . Die"
33,pcc2/11299,", ärgerlich ist es trotzdem . Und aberwitzig dazu .","Es entbehrt nicht der Komik , wenn sich drei Erwachsene - Karola Andrae ( Bürgerbündnis/FDP ) , Susanne Michler ( CDU ) und Joachim Zanow ( SPD ) - darüber streiten , was Jugendliche wollen und brauchen , ohne auf die Idee zu kommen ,",,sie,,selbst zu fragen .,"Und das , obwohl sie ihnen gegenüber sitzen . Die"
34,pcc2/11299,"die Idee zu kommen , sie selbst zu fragen .",Und,,das,,", obwohl sie ihnen gegenüber sitzen .",Die Jugendlichen wurden somit zum bloßen Feigenblatt degradiert . Nicht
35,pcc2/11299,"die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl",,sie,,ihnen gegenüber sitzen .,Die Jugendlichen wurden somit zum bloßen Feigenblatt degradiert . Nicht
36,pcc2/11299,"die Idee zu kommen , sie selbst zu fragen .","Und das , obwohl sie",,ihnen,,gegenüber sitzen .,Die Jugendlichen wurden somit zum bloßen Feigenblatt degradiert . Nicht
37,pcc2/11299,". Und das , obwohl sie ihnen gegenüber sitzen .",Die Jugendlichen,,wurden somit zum bloßen Feigenblatt degradiert .,Nicht über sondern mit ihnen hätten die Politiker reden sollen,,
38,pcc2/11299,". Und das , obwohl sie ihnen gegenüber sitzen .",Die Jugendlichen wurden somit zum,,bloßen Feigenblatt,,degradiert .,Nicht über sondern mit ihnen hätten die Politiker reden sollen
39,pcc2/11299,. Die Jugendlichen wurden somit zum bloßen Feigenblatt degradiert .,Nicht über sondern mit,,ihnen,,hätten die Politiker reden sollen .,Damit ist eine große Chance vertan . Vielleicht klappt es
40,pcc2/11299,. Die Jugendlichen wurden somit zum bloßen Feigenblatt degradiert .,Nicht über sondern mit ihnen hätten,,die Politiker,,reden sollen .,Damit ist eine große Chance vertan . Vielleicht klappt es
41,pcc2/11299,über sondern mit ihnen hätten die Politiker reden sollen .,Damit ist,,eine große Chance,,vertan .,Vielleicht klappt es bei der nächsten Runde Anfang 2002 .
42,pcc2/11299,reden sollen . Damit ist eine große Chance vertan .,Vielleicht klappt,,es,,bei der nächsten Runde Anfang 2002 .,Dann werden auch mehr Jugendliche eingeladen . In der Gruppe
43,pcc2/11299,reden sollen . Damit ist eine große Chance vertan .,Vielleicht klappt es bei,,der nächsten Runde,,Anfang 2002 .,Dann werden auch mehr Jugendliche eingeladen . In der Gruppe
44,pcc2/11299,reden sollen . Damit ist eine große Chance vertan .,Vielleicht klappt es bei der nächsten Runde,,Anfang 2002,,.,Dann werden auch mehr Jugendliche eingeladen . In der Gruppe
45,pcc2/11299,Vielleicht klappt es bei der nächsten Runde Anfang 2002 .,Dann werden auch,,mehr Jugendliche,,eingeladen .,In der Gruppe können sie sich hoffentlich mehr Gehör verschaffen
46,pcc2/11299,Anfang 2002 . Dann werden auch mehr Jugendliche eingeladen .,In,,der Gruppe,,können sie sich hoffentlich mehr Gehör verschaffen .,Vielleicht finden dann auch Vertreter von PDS und Gewerbeverein ihren
47,pcc2/11299,Anfang 2002 . Dann werden auch mehr Jugendliche eingeladen .,In der Gruppe können,,sie,,sich hoffentlich mehr Gehör verschaffen .,Vielleicht finden dann auch Vertreter von PDS und Gewerbeverein ihren
48,pcc2/11299,Anfang 2002 . Dann werden auch mehr Jugendliche eingeladen .,In der Gruppe können sie sich hoffentlich,,mehr Gehör,,verschaffen .,Vielleicht finden dann auch Vertreter von PDS und Gewerbeverein ihren
49,pcc2/11299,der Gruppe können sie sich hoffentlich mehr Gehör verschaffen .,Vielleicht finden dann auch,,Vertreter,,von PDS und Gewerbeverein ihren Weg ins Rathaus .,Die glänzten diesmal noch mit Abwesenheit .
50,pcc2/11299,der Gruppe können sie sich hoffentlich mehr Gehör verschaffen .,Vielleicht finden dann auch Vertreter von,,PDS,,und Gewerbeverein ihren Weg ins Rathaus .,Die glänzten diesmal noch mit Abwesenheit .
51,pcc2/11299,der Gruppe können sie sich hoffentlich mehr Gehör verschaffen .,Vielleicht finden dann auch Vertreter von PDS und,,Gewerbeverein,,ihren Weg ins Rathaus .,Die glänzten diesmal noch mit Abwesenheit .
52,pcc2/11299,der Gruppe können sie sich hoffentlich mehr Gehör verschaffen .,Vielleicht finden dann auch Vertreter von PDS und Gewerbeverein,,ihren Weg,,ins Rathaus .,Die glänzten diesmal noch mit Abwesenheit .
53,pcc2/11299,der Gruppe können sie sich hoffentlich mehr Gehör verschaffen .,Vielleicht finden dann auch Vertreter von PDS und Gewerbeverein ihren Weg ins,,Rathaus,,.,Die glänzten diesmal noch mit Abwesenheit .
54,pcc2/11299,Vertreter von PDS und Gewerbeverein ihren Weg ins Rathaus .,Die,,glänzten diesmal noch mit Abwesenheit,.,,
55,pcc2/11299,Vertreter von PDS und Gewerbeverein ihren Weg ins Rathaus .,Die glänzten diesmal noch mit,,Abwesenheit,.,,
56,pcc2/4282,Steilpass,Wunder,,gibt es immer wieder !,Erst spielen die Dallgower Gemeindevertreter so statisch und verzagt wie,,
57,pcc2/4282,Steilpass Wunder gibt es immer wieder !,Erst spielen,,die Dallgower Gemeindevertreter,,so statisch und verzagt wie die deutsche Abwehrreihe der Fußballkicker .,Und dann kommt aus der Tiefe solch ein fulminanter Steilpass
58,pcc2/4282,Steilpass Wunder gibt es immer wieder !,Erst spielen die Dallgower Gemeindevertreter so statisch und verzagt wie,,die deutsche Abwehrreihe der Fußballkicker,,.,Und dann kommt aus der Tiefe solch ein fulminanter Steilpass
59,pcc2/4282,statisch und verzagt wie die deutsche Abwehrreihe der Fußballkicker .,Und dann kommt aus,,der Tiefe,,"solch ein fulminanter Steilpass , von dem man hofft , dass die Seeburger oder Groß-Glienicker Mitspieler ihn aufnehmen können .","Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst"
60,pcc2/4282,statisch und verzagt wie die deutsche Abwehrreihe der Fußballkicker .,Und dann kommt aus der Tiefe solch,,ein fulminanter Steilpass,,", von dem man hofft , dass die Seeburger oder Groß-Glienicker Mitspieler ihn aufnehmen können .","Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst"
61,pcc2/4282,statisch und verzagt wie die deutsche Abwehrreihe der Fußballkicker .,"Und dann kommt aus der Tiefe solch ein fulminanter Steilpass , von",,dem,,"man hofft , dass die Seeburger oder Groß-Glienicker Mitspieler ihn aufnehmen können .","Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst"
62,pcc2/4282,statisch und verzagt wie die deutsche Abwehrreihe der Fußballkicker .,"Und dann kommt aus der Tiefe solch ein fulminanter Steilpass , von dem",,man,,"hofft , dass die Seeburger oder Groß-Glienicker Mitspieler ihn aufnehmen können .","Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst"
63,pcc2/4282,statisch und verzagt wie die deutsche Abwehrreihe der Fußballkicker .,"Und dann kommt aus der Tiefe solch ein fulminanter Steilpass , von dem man hofft , dass",,die Seeburger oder Groß-Glienicker Mitspieler,,ihn aufnehmen können .,"Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst"
64,pcc2/4282,statisch und verzagt wie die deutsche Abwehrreihe der Fußballkicker .,"Und dann kommt aus der Tiefe solch ein fulminanter Steilpass , von dem man hofft , dass die Seeburger oder Groß-Glienicker Mitspieler",,ihn,,aufnehmen können .,"Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst"
65,pcc2/4282,dass die Seeburger oder Groß-Glienicker Mitspieler ihn aufnehmen können .,Ein Befreiungsschlag,,"ist es allerdings nicht , weil es vorerst keine Gefahr fürs Dallgower Tor gab .",Die Seeburger und einige Groß-Glienicker haben den Ball erst zurückgespielt,,
66,pcc2/4282,dass die Seeburger oder Groß-Glienicker Mitspieler ihn aufnehmen können .,"Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst",,keine Gefahr,,fürs Dallgower Tor gab .,Die Seeburger und einige Groß-Glienicker haben den Ball erst zurückgespielt
67,pcc2/4282,dass die Seeburger oder Groß-Glienicker Mitspieler ihn aufnehmen können .,"Ein Befreiungsschlag ist es allerdings nicht , weil es vorerst keine Gefahr fürs",,Dallgower Tor,,gab .,Die Seeburger und einige Groß-Glienicker haben den Ball erst zurückgespielt
68,pcc2/4282,weil es vorerst keine Gefahr fürs Dallgower Tor gab .,Die Seeburger,,und einige Groß-Glienicker haben den Ball erst zurückgespielt und dann um so drängender wieder gefordert .,"Nun sollen sie zeigen , wie sie die Chance verwerten",,
69,pcc2/4282,weil es vorerst keine Gefahr fürs Dallgower Tor gab .,Die Seeburger und,,einige Groß-Glienicker,,haben den Ball erst zurückgespielt und dann um so drängender wieder gefordert .,"Nun sollen sie zeigen , wie sie die Chance verwerten"
70,pcc2/4282,weil es vorerst keine Gefahr fürs Dallgower Tor gab .,Die Seeburger und einige Groß-Glienicker haben,,den Ball,,erst zurückgespielt und dann um so drängender wieder gefordert .,"Nun sollen sie zeigen , wie sie die Chance verwerten"
71,pcc2/4282,erst zurückgespielt und dann um so drängender wieder gefordert .,Nun sollen,,sie,,"zeigen , wie sie die Chance verwerten .","Eine Diskussion , wo künftig die Trainerkabine stehen soll ,"
72,pcc2/4282,erst zurückgespielt und dann um so drängender wieder gefordert .,"Nun sollen sie zeigen , wie",,sie,,die Chance verwerten .,"Eine Diskussion , wo künftig die Trainerkabine stehen soll ,"
73,pcc2/4282,erst zurückgespielt und dann um so drängender wieder gefordert .,"Nun sollen sie zeigen , wie sie",,die Chance,,verwerten .,"Eine Diskussion , wo künftig die Trainerkabine stehen soll ,"
74,pcc2/4282,"sollen sie zeigen , wie sie die Chance verwerten .",Eine Diskussion,,", wo künftig die Trainerkabine stehen soll , wäre in der jetzigen Spielsituation verheerend .",Und eine Parallele zu den deutschen Grotten-Kickern gibt es immer,,
75,pcc2/4282,"sollen sie zeigen , wie sie die Chance verwerten .","Eine Diskussion , wo künftig",,die Trainerkabine,,"stehen soll , wäre in der jetzigen Spielsituation verheerend .",Und eine Parallele zu den deutschen Grotten-Kickern gibt es immer
76,pcc2/4282,"sollen sie zeigen , wie sie die Chance verwerten .","Eine Diskussion , wo künftig die Trainerkabine stehen soll , wäre in",,der jetzigen Spielsituation,,verheerend .,Und eine Parallele zu den deutschen Grotten-Kickern gibt es immer
77,pcc2/4282,"stehen soll , wäre in der jetzigen Spielsituation verheerend .",Und,,eine Parallele,,zu den deutschen Grotten-Kickern gibt es immer noch .,Auch wenn die Spieler aus den verschiedenen Vereinen zusammengewürfelt sind
78,pcc2/4282,"stehen soll , wäre in der jetzigen Spielsituation verheerend .",Und eine Parallele zu,,den deutschen Grotten-Kickern,,gibt es immer noch .,Auch wenn die Spieler aus den verschiedenen Vereinen zusammengewürfelt sind
79,pcc2/4282,Parallele zu den deutschen Grotten-Kickern gibt es immer noch .,Auch wenn,,die Spieler,,"aus den verschiedenen Vereinen zusammengewürfelt sind , sie müssen sich daran gewöhnen , dass sie nun in einer Mannschaft "" Döberitzer Heide "" spielen .",Und das heißt gemeinsam und nicht gegeneinander . Ermahnungen von
80,pcc2/4282,Parallele zu den deutschen Grotten-Kickern gibt es immer noch .,Auch wenn die Spieler aus,,den verschiedenen Vereinen,,"zusammengewürfelt sind , sie müssen sich daran gewöhnen , dass sie nun in einer Mannschaft "" Döberitzer Heide "" spielen .",Und das heißt gemeinsam und nicht gegeneinander . Ermahnungen von
81,pcc2/4282,Parallele zu den deutschen Grotten-Kickern gibt es immer noch .,"Auch wenn die Spieler aus den verschiedenen Vereinen zusammengewürfelt sind ,",,sie,,"müssen sich daran gewöhnen , dass sie nun in einer Mannschaft "" Döberitzer Heide "" spielen .",Und das heißt gemeinsam und nicht gegeneinander . Ermahnungen von
82,pcc2/4282,Parallele zu den deutschen Grotten-Kickern gibt es immer noch .,"Auch wenn die Spieler aus den verschiedenen Vereinen zusammengewürfelt sind , sie müssen",,sich,,"daran gewöhnen , dass sie nun in einer Mannschaft "" Döberitzer Heide "" spielen .",Und das heißt gemeinsam und nicht gegeneinander . Ermahnungen von
83,pcc2/4282,Parallele zu den deutschen Grotten-Kickern gibt es immer noch .,"Auch wenn die Spieler aus den verschiedenen Vereinen zusammengewürfelt sind , sie müssen sich daran gewöhnen , dass",,sie,,"nun in einer Mannschaft "" Döberitzer Heide "" spielen .",Und das heißt gemeinsam und nicht gegeneinander . Ermahnungen von
84,pcc2/4282,Parallele zu den deutschen Grotten-Kickern gibt es immer noch .,"Auch wenn die Spieler aus den verschiedenen Vereinen zusammengewürfelt sind , sie müssen sich daran gewöhnen , dass sie nun in",,"einer Mannschaft "" Döberitzer Heide """,,spielen .,Und das heißt gemeinsam und nicht gegeneinander . Ermahnungen von
85,pcc2/4282,"nun in einer Mannschaft "" Döberitzer Heide "" spielen .",Und,,das,,heißt gemeinsam und nicht gegeneinander .,"Ermahnungen von der Seitenlinie , miteinander fair umzugehen und sich"
86,pcc2/4282,spielen . Und das heißt gemeinsam und nicht gegeneinander .,Ermahnungen,,"von der Seitenlinie , miteinander fair umzugehen und sich nicht beim kleinsten Schubser gegenseitig zu zerfleischen , sind normalerweise überflüssig .",Vorerst allerdings hilfreich .,,
87,pcc2/4282,spielen . Und das heißt gemeinsam und nicht gegeneinander .,Ermahnungen von,,der Seitenlinie,,", miteinander fair umzugehen und sich nicht beim kleinsten Schubser gegenseitig zu zerfleischen , sind normalerweise überflüssig .",Vorerst allerdings hilfreich .
88,pcc2/4282,spielen . Und das heißt gemeinsam und nicht gegeneinander .,"Ermahnungen von der Seitenlinie , miteinander fair umzugehen und",,sich,,"nicht beim kleinsten Schubser gegenseitig zu zerfleischen , sind normalerweise überflüssig .",Vorerst allerdings hilfreich .
89,pcc2/4282,spielen . Und das heißt gemeinsam und nicht gegeneinander .,"Ermahnungen von der Seitenlinie , miteinander fair umzugehen und sich nicht beim",,kleinsten Schubser,,"gegenseitig zu zerfleischen , sind normalerweise überflüssig .",Vorerst allerdings hilfreich .

